tags/TagMap.md

```diff
@@ -23,23 +23,78 @@ and group stats, annotate traces and logs.
-propagation when the call leaves the process.
-
-**Request Scope**
-If a tag is created with the `Request` scope then it is propagated across process boundaries subject 
-to outgoing and incoming (on remote side) filter criteria. See `TagPropagationFilter` in 
-[Tag Propagation](#Tag Propagation). Typically 'Request' tags represents a request, processing
- of which may span multiple entities.
+## TagMetadata
+
+`TagMetadata` contains properties associated with a `Tag`. For now only the property `TagTTL`
+is defined. In future, additional properties may be added to address specific situations.
+
+A tag creator determines metadata of a tag it creates.
+
+### TagTTL
+
+`TagTTL` is an integer that represents number of hops a tag can propagate. Anytime a sender serializes a tag,
+sends it over the wire and receiver unserializes the tag then the tag is considered to have travelled one hop. 
+There could be one or more proxy(ies) between sender and receiver. Proxies are treated as transparent
+entities and they do not create additional hops. Every propagation implementation shoud support option 
+`decrementTTL` (default set to true) that allows proxies to set it to false.
+
+**For now, ONLY special values (0 and -1) are supported.**
+
+#### Special Values
+- **NO_PROPAGATION (0)**: Tag with `TagTTL` value of zero is considered to have local scope and
+ is used within the process it created.
+ 
+- **UNLIMITED_PROPAGATION (-1)**: A Tag with `TagTTL` value of -1 can propagate unlimited hops.
+ However, it is still subject to outgoing and incoming (on remote side) filter criteria. 
+ See `TagPropagationFilter` in [Tag Propagation](#Tag Propagation). `TagTTL` value of -1
+ is typical used to represent a request, processing of which may span multiple entities.
+
+#### Example for TagTTL > 0
+On a server side typically there is no information about the caller besides ip/port,
+but in every process there is a notion of "service_name" tag that is added as a "caller" tag before
+serialization when a RPC/HTTP call is made. For the "caller" tag, desirable `TagTTL` value is 1.
+
+Note that TagTTL value of 1 is not supported at this time. The example is listed here simply to
+show a possible use case for TagTTL > 0.
+ 
+### Processing at Receiver and Sender
+For the sake of completeness, processing of `Tag` and `TagTTL` at sender and receiver includes
+the values other than the special values of `TagTTL`.
+
+#### At Receiver
+Upon receiving a tag from remote entity a tag extractor
+
+- MUST decrement the value of `TagTTL` by one if it is greater than zero.
+- MUST not change the value of `TagTTL` if it is -1.
+- MUST treat the value of `TagTTL` as -1 if is not present.
+- MUST discard the `Tag` for any other value of `TagTTL`.
+
+#### At Sender
+Upon preparing to send a tag to a remote entity a tag injector
+- MUST send the tag AND include `TagTTL` if its value is greater than 0.
```

Comment:
We don't have defined a tag-metadata in our binary format, should we consider to not send this for the moment, or send it as TTL -1 in this phase?

Reply:
Again this for future. Today nothing goes on wire. When we implement TagTTL>0 then we would need to define tag-metadata for binary format which will include TagTTL.